src/plugins/home/server/services/sample_data/sample_data_installer.test.ts

```diff
@@ -0,0 +1,331 @@
+    soClient = savedObjectsClientMock.create();
+    soImporter = savedObjectsServiceMock.createImporter();
+    logger = loggingSystemMock.createLogger();
+
+    installer = new SampleDataInstaller({
+      esClient,
+      soClient,
+      soImporter,
+      logger,
+      sampleDatasets: testDatasets,
+    });
+
+    soImporter.import.mockResolvedValue({
+      success: true,
+      successCount: 1,
+      errors: [],
+      warnings: [],
+    });
+
+    soClient.delete.mockResolvedValue({});
+
+    esClient.asCurrentUser.indices.getAlias.mockImplementation(() => {
+      throw new Error('alias not found');
+    });
+
+    findSampleObjectsMock.mockResolvedValue([]);
+  });
+
+  afterEach(() => {
+    insertDataIntoIndexMock.mockReset();
+    findSampleObjectsMock.mockReset();
+  });
+
+  describe('#install', () => {
+    it('cleanups the data index before installing', async () => {
+      await installer.install('test_single_data_index');
+
+      expect(esClient.asCurrentUser.indices.delete).toHaveBeenCalledTimes(1);
+      expect(esClient.asCurrentUser.indices.delete).toHaveBeenCalledWith({
+        index: 'kibana_sample_data_test_single_data_index',
+      });
+    });
+
+    it('creates the data index', async () => {
+      await installer.install('test_single_data_index');
+
+      expect(esClient.asCurrentUser.indices.create).toHaveBeenCalledTimes(1);
+      expect(esClient.asCurrentUser.indices.create).toHaveBeenCalledWith({
+        index: 'kibana_sample_data_test_single_data_index',
+        body: {
+          settings: { index: { number_of_shards: 1, auto_expand_replicas: '0-1' } },
+          mappings: { properties: { someField: { type: 'keyword' } } },
+        },
+      });
+    });
+
+    it('inserts the data into the index', async () => {
+      await installer.install('test_single_data_index');
+
+      expect(insertDataIntoIndexMock).toHaveBeenCalledTimes(1);
+      expect(insertDataIntoIndexMock).toHaveBeenCalledWith({
+        index: 'kibana_sample_data_test_single_data_index',
+        nowReference: expect.any(String),
+        logger,
+        esClient,
+        dataIndexConfig: testDatasets[0].dataIndices[0],
+      });
+    });
+
+    it('imports the saved objects', async () => {
+      await installer.install('test_single_data_index');
+
+      expect(soImporter.import).toHaveBeenCalledTimes(1);
+      expect(soImporter.import).toHaveBeenCalledWith({
+        readStream: expect.any(Readable),
+        overwrite: true,
+        createNewCopies: false,
+      });
+    });
+
+    it('throws a SampleDataInstallError with code 404 when the dataset is not found', async () => {
+      try {
+        await installer.install('unknown_data_set');
+        expect('should have returned an error').toEqual('but it did not');
+      } catch (e) {
+        expect(e).toBeInstanceOf(SampleDataInstallError);
+        expect((e as SampleDataInstallError).httpCode).toEqual(404);
+      }
```

Comment:
Unless I'm wrong, `.rejects.toThrowError(new SampleDataInstallError(...))` will assert against the error's `msg` too, so I would like to recreate a `SampleDataInstallError` with the same message we're throwing in the actual code, which I wanted to avoid